python/mxnet/contrib/autograd.py

```diff
@@ -10,15 +10,70 @@
 from ..ndarray import NDArray, zeros_like
 from ..symbol import _GRAD_REQ_MAP
 
-def set_recording(recording):
-    """Turn on or turn off operator recording.
+
+def set_is_training(is_train):
+    """Set status to training/not training. When training, graph will be constructed
+    for gradient computation. Operators will also run with ctx.is_train=True. For example,
+    Dropout will drop inputs randomly when is_train=True while simply passing through
+    if is_train=False.
 
     Parameters
     ----------
-    recording: bool
+    is_train: bool
+
+    Returns
+    -------
+    previous state before this set.
+    """
+    prev = ctypes.c_int()
+    check_call(_LIB.MXAutogradSetIsTraining(
+        ctypes.c_int(is_train), ctypes.byref(prev)))
+    return bool(prev.value)
+
+
+class TrainingStateScope(object):
+    """Scope for managing training state.
+
+    Example::
+        with TrainingStateScope(True):
+            y = model(x)
+            compute_gradient([y])
```

Comment:
I think `compute_gradient` should not be put in the training scope? The example may need to be changed.